README.md

```diff
@@ -26,26 +26,237 @@ the instructions below:
   - select the Knot.x panel
 
 ## How does it work?
-Knot.x working in debug mode marks the beginning and the end of the snippet fragment with HTML 
-comments:
+
+### Debug fragment data
+
+Knot.x Fragments, when run in debug mode, injects information about fragments into the output.
+This information can then be read, parsed and displayed by various tools. Knot.x Fragments Chrome Extension is an official tool dedicated to this very purpose.
+
+Of course fragments' outputs can come in a myriad of formats. Right now Knot.x supports injecting debug information into HTML only. 
+Therefore, this tool only works for fragment-generated HTML pages. 
+JSON support is planned for the near future. 
+
+### HTML structure
+
+When an HTML document is generated by the Fragments Engine in a debug mode, there are special comments and tags injected in order to mark what data came from which fragment.
+
+For example, a page like this:
+
+```html
+<body>
+  <header>
+    Weather Service
+  </header>
+  
+  <knotx:snippet data-knotx-task="weather-task">
+    <span>
+      {{fetch-weather-info._result.temperatue}} in {{fetch-weather-info._result.location}}
+    </span>
+  </knotx:snippet>
+</body>
 ```
-<!-- data-knotx-id="FRAGMENT_IDENTIFIER" -->
-  FRAGMENT_BODY
-<!-- data-knotx-id="FRAGMENT_IDENTIFIER" -->
+
+Could result in:
+
+```html
+<body>
+  <header>
+    Weather Service
+  </header>
+  
+  <!-- data-knotx-id="auto-generated-id" -->
+  <script data-knotx-debug="log" data-knotx-id="auto-generated-id" type="application/json">{ debug-data-here }</script>
+
+  <span>
+    -2°C in Reykjavík
+  </span>
+
```

Comment:
Do we need this empty line here?

Reply:
Same logic as with the previous empty line